lib/tanker/identity.rb

```diff
@@ -1,6 +1,55 @@
 require 'tanker/crypto'
+require 'base64'
+require 'json'
 
 module Tanker
   module Identity
+    class InvalidIdentity < StandardError; end
+
+    BLOCK_HASH_SIZE = 32
+    USER_SECRET_SIZE = 32
+
+    private
+
+    def self.hash_user_id(trustchain_id, user_id)
+      binary_user_id = user_id.dup.force_encoding(Encoding::ASCII_8BIT)
+      Crypto.generichash(binary_user_id + trustchain_id, BLOCK_HASH_SIZE)
+    end
+
+    def self.user_secret(hashed_user_id)
+      random_bytes = Crypto.random_bytes(USER_SECRET_SIZE - 1)
+      check = Crypto.generichash(random_bytes + hashed_user_id, Crypto::HASH_MIN_SIZE)
+      random_bytes + check[0]
+    end
+
+    public
+
+    def self._deserialize(b64_json)
```

Comment:
remove this underscore, I don't think it hurts

Reply:
Fixed